tests/test_tzinfo.py

```diff
@@ -0,0 +1,207 @@
+import copy
+import functools
+import pickle
+import unittest
+from datetime import datetime, timedelta, timezone, tzinfo
+
+from pydantic_core import SchemaValidator, TzInfo, core_schema
+
+
+class _ALWAYS_EQ:
+    """
+    Object that is equal to anything.
+    """
+
+    def __eq__(self, other):
+        return True
+
+    def __ne__(self, other):
+        return False
+
+
+ALWAYS_EQ = _ALWAYS_EQ()
+
+
+@functools.total_ordering
+class _LARGEST:
+    """
+    Object that is greater than anything (except itself).
+    """
+
+    def __eq__(self, other):
+        return isinstance(other, _LARGEST)
+
+    def __lt__(self, other):
+        return False
+
+
+LARGEST = _LARGEST()
+
+
+@functools.total_ordering
+class _SMALLEST:
+    """
+    Object that is less than anything (except itself).
+    """
+
+    def __eq__(self, other):
+        return isinstance(other, _SMALLEST)
+
+    def __gt__(self, other):
+        return False
+
+
+SMALLEST = _SMALLEST()
+
+
+pickle_choices = [(pickle, pickle, proto) for proto in range(pickle.HIGHEST_PROTOCOL + 1)]
+
+HOUR = timedelta(hours=1).total_seconds()
+ZERO = timedelta(0).total_seconds()
+
+
+def first_sunday_on_or_after(dt):
+    days_to_go = 6 - dt.weekday()
+    if days_to_go:
+        dt += timedelta(days_to_go)
+    return dt
+
+
+DSTSTART = datetime(1, 4, 1, 2)
+DSTEND = datetime(1, 10, 25, 1)
+
+
+class TestTzInfo(unittest.TestCase):
```

Comment:
assuming this is copied from cpython somewhere, please include a link to the original in a docstring.